extensions/Worker.Extensions.Http.AspNetCore/src/AspNetMiddleware/InvokeFunctionMiddleware.cs

```diff
@@ -3,23 +3,27 @@
 
 using System.Threading.Tasks;
 using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Primitives;
 
 namespace Microsoft.Azure.Functions.Worker.Extensions.Http.AspNetCore
 {
     internal class InvokeFunctionMiddleware
     {
-        private readonly IHttpCoordinator _coordinator;
-
-        public InvokeFunctionMiddleware(RequestDelegate next, IHttpCoordinator httpCoordinator)
+        public InvokeFunctionMiddleware(RequestDelegate next)
         {
-            _coordinator = httpCoordinator;
         }
 
         public Task Invoke(HttpContext context)
         {
+            return InvokeInternal(context);
+        }
+
+        internal static Task InvokeInternal(HttpContext context)
+        {
+            var coordinator = context.RequestServices.GetRequiredService<IHttpCoordinator>();
```

Comment:
I needed a delegate to pass to the endpoint...

Reply:
I see. I don't know if this is the best place to share that delegate. From a reader's perspective, it is not clear why a delegate from `InvokeFunctionMiddleware` would be used as the implementation of a `RouteEndpoint`.

Maybe an internal extension method?

``` CSharp
internal static Task InvokeFunctionAsync(this HttpContext context)
{
    var coordinator = context.RequestServices.GetRequiredService<IHttpCoordinator>();
    context.Request.Headers.TryGetValue(Constants.CorrelationHeader, out StringValues invocationId);
    return coordinator.RunFunctionInvocationAsync(invocationId);
}
```

Or honestly, remaining DRY here may not even be that crucial. I wouldn't be opposed to copy & pasting this code.

Reply:
I'm for the code reuse, so I'd avoid duplicating this code. I'd like to avoid sprinkling this correlation logic in multiple places.

I like the extension method suggestion, but also wondering if there's an opportunity for us to enhance the `IHttpCoordinator` API to abstract away correlation logic handling and work directly with the context, but if we decided to take that route, it should be outside of this PR